File: lib/compile_systems/go_system.cpp
#include <iostream>
#include <fstream>
using namespace std;

char command[1024];

int main(int argc, char const *argv[])
{
  ofstream ofs;
  ofs.open("main.go",ios::binary);
  string buf;
  string splitter="<$><*><$>";
  string time;
  while(getline(cin,buf)){
    if(buf==splitter)break;
    ofs<<buf<<endl;
  }
  ofs.close();
  ofs.open("test.in",ios::binary);
  while(getline(cin,buf)){
    if(buf==splitter)break;
    ofs<<buf<<endl;
  }
  ofs.close();
  cin>>time;
  system(("timeout -s 9 " + time + " /opt/wandbox/go-head/bin/go run main.go < test.in ").c_str());
  return 0;
}

File: lib/compile_systems/go_system.rb
require 'base64'

File.open('stdin.txt.bz2', 'wb') do |file|
  file.write Base64.decode64 $stdin.read
  file.close
end

system('bunzip2 stdin.txt.bz2')

stdin = File.open('stdin.txt').read.split("\n<$><*><$>\n")
code = stdin[0]
input = stdin[1]
time = stdin[2]

File.open('./main.go', 'w') do |file|
  file.puts code
  file.close
end

File.open('./test.in', 'w') do |file|
  file.puts input
  file.close
end

system('timeout -s 9 ' + time + ' /opt/wandbox/go-head/bin/go run main.go < test.in')
